red/MyDate/lib/MyDate.pm

```diff
@@ -0,0 +1,169 @@
+package MyDate;
+
+use v5.16;
+use strict;
+use warnings;
+use Carp;
+our $AUTOLOAD;
+
+=head1 NAME
+
+MyDate - Yet another simple utility for dates
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+
+=head1 SYNOPSIS
+
+Simple OO interface onto localtime.
+
+    use MyDate;
+
+    my $date = MyDate->new();
+    my ( $day, $month, $year ) =
+      $date->date, $date->month, $date->year; # e.g. 26 4 2015
+    ...
+
+=head1 SUBROUTINES/METHODS
+
+=head2 new
+
+Get an instance of MyDate. Not a whole lot to configure here.
+
+=cut
+
+sub new {
+  bless \my $self, $_[0]
```

Comment:
The `\my` looks all sorts of wrong. It might be more readable if you had either
```perl
\( my $self )
```
or 
```perl
\ my $self
```
in order to make it clearer exactly what you're applying the reference operator to.